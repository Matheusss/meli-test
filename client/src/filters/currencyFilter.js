import Vue from 'vue'

Vue.filter('currency', (value) => {
  if (typeof value !== 'number') {
    return value
  }

  const formatter = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    minimumFractionDigits: 2,
  })

  return formatter.format(value)
})
