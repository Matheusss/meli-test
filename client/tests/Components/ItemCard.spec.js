import { shallowMount } from '@vue/test-utils'
import Component from '@/modules/Items/submodules/List/components/item-card/item-card.vue'
import currencyFilter from '@/filters/currencyFilter'
import VueRouter from 'vue-router'
import Vue from 'vue'

describe('ItemsCard', () => {
  let wrapper
  const instance = Vue.extend()
  
  beforeEach(() => {
    Vue.use(VueRouter)
    const router = new VueRouter({routes: [
      {path: '/items', name: 'items', component: {render: h => '-'}},
      {path: '/items/:id', name: 'items.details', component: {render: h => '-'}},
    ]})

    wrapper = shallowMount(Component, {
      instance,
      router,
      filter: currencyFilter,
      propsData: {
        item: {
          id: 1,
          title: 'Component title',
          price: {
            ammount: 1,
            currency: 'USD',
            decimals: 123,
          },
          picture: '',
          condition: 'new',
          free_shipping: false,
        }
      }
    })
  })

  it('should be a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('should renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should navigate to item details route after search', () => {
    const spy = jest.spyOn(wrapper.vm, 'navigateToDetails')
    wrapper.vm.navigateToDetails()
    expect(spy).toHaveBeenCalled()
    expect(wrapper.vm.$route.name).toBe('items.details')
    expect(wrapper.vm.$route.params).toEqual({ id: 1})
  })
})
