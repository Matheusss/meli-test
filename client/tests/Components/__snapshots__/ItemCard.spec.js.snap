// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader should renders correctly 1`] = `undefined`;

exports[`ItemsCard should renders correctly 1`] = `
<div class="item-list-card"><img src="" alt="Component title" height="180" width="180">
  <div class="item-infos">
    <p class="item-price">
      $123.00
    </p>
    <p>Component title</p>
    <p>new</p>
  </div>
</div>
`;
